src/DocScan/Session/Retrieve/Configuration/Capture/RequiredResourceResponse.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Yoti\DocScan\Session\Retrieve\Configuration\Capture;
+
+use Yoti\DocScan\Constants;
+use Yoti\DocScan\Session\Retrieve\Configuration\Capture\Source\AllowedSourceResponse;
+use Yoti\DocScan\Session\Retrieve\Configuration\Capture\Source\EndUserAllowedSourceResponse;
+use Yoti\DocScan\Session\Retrieve\Configuration\Capture\Source\IbvAllowedSourceResponse;
+use Yoti\DocScan\Session\Retrieve\Configuration\Capture\Source\RelyingBusinessAllowedSourceResponse;
+use Yoti\DocScan\Session\Retrieve\Configuration\Capture\Source\UnknownAllowedSourceResponse;
+
+class RequiredResourceResponse
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $state;
+
+    /**
+     * @var array<int,AllowedSourceResponse>
+     */
+    private $allowedSources;
+
+    /**
+     * @param array<string, mixed> $captureData
+     */
+    public function __construct(array $captureData)
+    {
+        $this->type = $captureData['type'] ?? null;
+        $this->id = $captureData['id'] ?? null;
+        $this->state = $captureData['state'] ?? null;
+
+        if (isset($captureData['allowed_sources'])) {
+            foreach ($captureData['allowed_sources'] as $allowedSource) {
+                $this->allowedSources[] = $this->createAllowedSourceFromArray($allowedSource);
+            }
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getState(): string
+    {
+        return $this->state;
+    }
+
+    /**
+     * @return AllowedSourceResponse[]
+     */
+    public function getAllowedSources(): array
+    {
+        return $this->allowedSources;
+    }
+
+    /**
+     *
+     * Returns if the Relying Business is allowed to upload resources
+     * to satisfy the requirement.
+     *
+     * return the end user is allowed to upload resources
+     *
+     * @return bool
+     */
+    public function isRelyingBusinessAllowed(): bool
+    {
+        if (($this->allowedSources != null)) {
```

Comment:
Sonar flagged these brackets as being unnecessary - https://sonarcloud.io/project/issues?id=getyoti%3Aphp&open=AX0EBJXKZNdrMs84Qdmx&pullRequest=241&resolved=false&types=CODE_SMELL
```suggestion
        if ($this->allowedSources != null) {
```